src/Kestrel.Core/Internal/HttpConnectionMiddleware.cs

```diff
@@ -54,6 +57,7 @@ public Task OnConnectionAsync(ConnectionContext connectionContext)
             };
 
             var connectionFeature = connectionContext.Features.Get<IHttpConnectionFeature>();
+            var lifetimeFeature = connectionContext.Features.Get<IConnectionLifetimeFeature>();
```

Comment:
This can be null right?

Reply:
No. Look at https://github.com/aspnet/KestrelHttpServer/pull/2619/files#diff-a6a566e69107e6ac28bc8c57908bf1f5L53.